### PR TITLE
[IBCDPE-419] Initial Setup of `pre-commit` for ADT

### PR DESCRIPTION
**Pull request by BWMac**
**Problem:**

We do not have `pre-commit` configured for the repository to enforce standards in formatting.

**Solution:**

Add and configure `pre-commit` for ADT. Add a step to our CI GitHub Action that runs it automatically.

Notes:
- I have added a number of `pre-commit` hooks, and committed the automatic changes that they applied to the codebase. There was no impact on tests passing or data processing completing as expected.
- Two of the `pre-commit` hooks that I added (`flake8` and `mypy`) I have left commented out in the configuration file. This is because these hooks do not automatically make changes for you. The number of changes that they suggested for them to pass was very large, and so if we want to add those hooks we should create separate tickets to make those changes.
- I updated `CONTRIBUTING.md` with information on how to run the `pre-commit` hooks locally.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+        args: ["--fix=auto"] # replace 'auto' with 'lf' to enforce Linux/Mac line endings or 'crlf' for Windows
+
+  - repo: https://github.com/PyCQA/autoflake
+    rev: v2.0.0
+    hooks:
+      - id: autoflake
+
+  - repo: https://github.com/pycqa/isort
```
Comment (thomasyu888) [inline .pre-commit-config.yaml]: Nit: try using ruff instead of isort

Comment (BWMac) [reply .pre-commit-config.yaml]: Added this change.

File: src/agoradatatools/etl/transform/gene_info.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def transform_gene_info(
     resource_url_suffix = "%22%5D%7D%5D%7D"
     tep_info["resource_url"] = tep_info.apply(
         lambda row: resource_url_prefix + row["hgnc_symbol"] + resource_url_suffix
-        if (row["is_adi"] == True) | (row["is_tep"] == True)
```
Comment (BWMac) [inline src/agoradatatools/etl/transform/gene_info.py:104]: Just want to draw some attention to this change. I don't think it changes the behavior of this function.

File: src/agoradatatools/etl/transform/gene_info.py [hunk 3]
```diff
@@ -101,7 +101,7 @@ def transform_gene_info(
...
+        if row["is_adi"] or row["is_tep"]
```
Comment (BWMac) [inline src/agoradatatools/etl/transform/gene_info.py:104]: Just want to flag this change. I don't think it changes the behavior of this function.

**Review by thomasyu888 — APPROVED** [review]
🔥 LGTM!

**Review by JessterB — APPROVED** [review]
lgtm!

**Pull request merged**